pkg/minikube/cluster/cluster.go

```diff
@@ -225,6 +229,21 @@ func UpdateCluster(h sshAble, d drivers.Driver, config KubernetesConfig) error {
 		}
 	}
 
+	// loop over .minikube/addons and transfer them to vm
+	searchDir := constants.MakeMiniPath("addons")
+	fileList := []string{}
+	err = filepath.Walk(searchDir, func(path string, f os.FileInfo, err error) error {
```

Comment:
Could you add some tests to make sure everything gets added correctly? It looks like you're handling nested directories as well, so make sure to include some examples.

An e2e test of this would be great too, but that can come later.